### PR TITLE
Adding Joi config checks to frontend

### PR DESCRIPTION
**Pull request by StanHus**
Relation: https://eaflood.atlassian.net/browse/FFAFGB-103

Adding checks for frontend using Joi

### CODE REVIEW

File: app/config/index.js [hunk 1]
```diff
@@ -1,15 +1,82 @@
+const Joi = require('joi')
+
 if (process.env.NODE_ENV !== 'test') {
   require('dotenv').config()
 }
 
+const schema = Joi.object({
+  env: Joi.string().valid('development', 'test', 'production').required(),
+
+  fundingFarmingApiUri: Joi.string().uri().required(),
```
Comment (MohammedFoysal) [inline app/config/index.js:10]: This env var is no longer used and can be removed. Remove it from here and the config on line 75

File: app/config/index.js [hunk 2]
```diff
@@ -1,15 +1,82 @@
...
+  fundingFarmingApiKey: Joi.string().required(),
```
Comment (MohammedFoysal) [inline app/config/index.js:11]: This one can be removed too

File: app/config/index.js [hunk 3]
```diff
@@ -1,15 +1,82 @@
...
+  appInsightsKey: Joi.string().optional(),
```
Comment (MohammedFoysal) [inline app/config/index.js:12]: This looks good. please edit the docker-compose.yaml file and remove the line where APPINSIGHTS_CONNECTIONSTRING is set then this check shouldn't give false positives

**Comment by MohammedFoysal** [comment]
Looks good, just two changes please. Also the unit tests are failing resulting in a failing build - can you investigate?

File: app/config/index.js (hunk 2 above)
Comment (StanHus) [reply app/config/index.js:11]: It seems to be used in 2 places actually. Still remove it?

Screenshot 2024-06-24 at 14 34 48

Comment (MohammedFoysal) [reply app/config/index.js:11]: Yes, if you can remove the usages too. We don't use this environment variable after switching to langchain. FARMING_FUNDING_API_URI can be removed too

File: app/config/index.js (hunk 3 above)
Comment (StanHus) [reply app/config/index.js:9]: done

File: app/config/index.js (hunk 2 above)
Comment (StanHus) [reply app/config/index.js]: done

**Comment by StanHus** [comment]
> Looks good, just two changes please. Also the unit tests are failing resulting in a failing build - can you investigate?

It was the mock server setup for the tests
I added more checks before launching the server, and it were those checks not letting the server run without additional env variables that would ensure the checks I added passed

**Review by MohammedFoysal — DISMISSED** [review]

**Review by MohammedFoysal — APPROVED** [review]

**Review by MohammedFoysal — APPROVED** [review]

**Review by MohammedFoysal — APPROVED** [review]

**Pull request merged**